File: backend/src/main/scala/microservices/traveler/objects/TravelerPreferenceInfo.scala
// 本文件定义 traveler 模块的偏好传输对象 `TravelerPreferenceInfo`，仅负责数据结构与 JSON codec。
package com.typesafe.travel.traveler.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TravelerPreferenceInfo(
    seatPreference: String,
    mealPreference: String,
    quietSeatPreferred: Boolean
)

object TravelerPreferenceInfo:
  given sourceEncoder: Encoder[TravelerPreferenceInfo] = deriveEncoder
  given sourceDecoder: Decoder[TravelerPreferenceInfo] = deriveDecoder

File: frontend/src/microservices/traveler/objects/TravelerPreferenceInfo.ts
// 本文件定义 traveler 模块的 `TravelerPreferenceInfo`，作为偏好信息传输数据并提供 JSON 编解码。

export type TravelerPreferenceInfo = {
  seatPreference: string
  mealPreference: string
  quietSeatPreferred: boolean
}

export const travelerPreferenceInfoFromJson = (json: string): TravelerPreferenceInfo =>
  JSON.parse(json) as TravelerPreferenceInfo

export const travelerPreferenceInfoToJson = (value: TravelerPreferenceInfo): string =>
  JSON.stringify(value)
